### PR TITLE
Reflect changes in draft-ietf-netconf-http-client-server draft

### LINKED ISSUE #47: Update the model to use http-client-grouping
https://datatracker.ietf.org/doc/html/draft-ietf-netconf-https-notif-15#section-6.2 shows that "grouping https-receiver-grouping” uses "http-client-stack-grouping”, which was removed in the latest version of that draft…the update that addresses the HTTP-client configuration issue (of the URL encoding most of the stack’s configuration).

The https-notif draft should be updatesd to use “grouping http-client-grouping” directly now.

### PR DESCRIPTION
**Pull request by mjethanandani**
These changes are by no means complete. However, ...

The first set of changes are trying to address the use of 'http-client-grouping' directly from ietf-http-client module. But is the mapping from what used to be 'remote-address' and 'remote-port' to 'uri' correct? @kwatsen, please examine.

Also, I do not know what 'local-definition' as part of 'tls-client-parameters' in example-https-notif-7.1.xml maps to anymore. Need your help there.

### CODE REVIEW

File: src/yang/ietf-https-notif-transport.yang [hunk 1]
```diff
@@ -87,27 +87,7 @@ module ietf-https-notif-transport {
...
-             path the receiver must support both the 'capabilities'
-             and 'relay-notification' resource targets, as described
-             in RFC XXXX.";
-        }
-        description
-          "Augmentation to add a receiver-specific path for the
-           'capabilities' and 'relay-notification' resources.";
-      }
-    }
+    uses httpc:http-client-grouping;
```
Comment (kwatsen) [inline src/yang/ietf-https-notif-transport.yang]: I think the `augment "transport/tls/tls/http-client-parameters" {` line needs to be inside this `uses` statement.  

Maybe `augment "/"` or maybe `augment ""` ?

In any casse, the `path` leaf needs to be present...

**Review by kwatsen — CHANGES_REQUESTED** [review]
In addition to the comment I left, I'm surprised that there was no updates to examples - did the the validation scripts throw errors?

File: src/yang/ietf-https-notif-transport.yang (hunk 1 above)
Comment (mjethanandani) [reply src/yang/ietf-https-notif-transport.yang]: Hmm. Does not like "/", "", or "./". pyang gives similar errors:

`../bin/ietf-https-notif-transport@2024-08-13.yang:91: error: bad value "./" (should be schema-nodeid)`

The http-client-grouping has a `uri` defined in the grouping. Any reason that cannot be used to specify the `path`?

Comment (kwatsen) [reply src/yang/ietf-https-notif-transport.yang]: `uri` is a `leaf`, so no way to augment a descendent into it!

Ugh, I don't have a good answer...

Maybe try "." for the path?

Comment (mjethanandani) [reply src/yang/ietf-https-notif-transport.yang]: Does not like "." either.

My question was, why can't the `path` be incorporated in the `uri` (not as a descendant node) just as `remote-address` and `remote-port` have been? Demonstrating here with an example.

OLD:
```
<remote-address>receiver.example.com</remote-address>
<remote-port>443</remote-port>
<path>some-path</path>

```
NEW:
`<uri>receiver.example.com:443/some-path</uri>`

Comment (kwatsen) [reply src/yang/ietf-https-notif-transport.yang]: ah, now I understand your proposal.

I like it, but I think the full fix would require two `uri` leafs, one for the 'capabilities' resource target, and the other for the 'relay-notification' resource target.

File: src/download-dependent-models.py [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+[ ["ietf-crypto-types", "draft-ietf-netconf-crypto-types", "34"],
+  ["ietf-truststore", "draft-ietf-netconf-trust-anchors", "28"],
+  ["ietf-keystore", "draft-ietf-netconf-keystore", "35"],
+  ["ietf-tcp-common", "draft-ietf-netconf-tcp-client-server", "26"],
+  ["ietf-tcp-client", "draft-ietf-netconf-tcp-client-server", "26"],
+  ["ietf-tls-client", "draft-ietf-netconf-tls-client-server", "41"],
+  ["ietf-tls-common", "draft-ietf-netconf-tls-client-server", "41"],
+  ["ietf-tls-server", "draft-ietf-netconf-tls-client-server", "41"],
+  ["iana-tls-cipher-suite-algs", "draft-ietf-netconf-tls-client-server", "41"],
+  ["ietf-http-client", "draft-ietf-netconf-http-client-server", "21"] ]
```
Comment (kwatsen) [inline src/download-dependent-models.py]: Now -23  (http-client-server-23)

File: src/download-dependent-models.py [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-  ["ietf-http-client", "draft-ietf-netconf-http-client-server", "11"] ]
+[ ["ietf-crypto-types", "draft-ietf-netconf-crypto-types", "34"],
+  ["ietf-truststore", "draft-ietf-netconf-trust-anchors", "28"],
+  ["ietf-keystore", "draft-ietf-netconf-keystore", "35"],
+  ["ietf-tcp-common", "draft-ietf-netconf-tcp-client-server", "26"],
+  ["ietf-tcp-client", "draft-ietf-netconf-tcp-client-server", "26"],
+  ["ietf-tls-client", "draft-ietf-netconf-tls-client-server", "41"],
+  ["ietf-tls-common", "draft-ietf-netconf-tls-client-server", "41"],
+  ["ietf-tls-server", "draft-ietf-netconf-tls-client-server", "41"],
+  ["iana-tls-cipher-suite-algs", "draft-ietf-netconf-tls-client-server", "41"],
```
Comment (kwatsen) [inline src/download-dependent-models.py:6]: What is this draft?  Should this line be removed given the following?
  - this draft doesn't depend on "netconf-client-server" draft
  - already the previous line points to the "tis-client-server" draft

File: src/yang/example-custom-module.yang [hunk 4]
```diff
@@ -35,14 +34,14 @@ module example-custom-module {
...
           "A list of HTTPS notif receivers.";
-       leaf name {
-         type string;
-         description
-           "A unique name for the https notif receiver.";
-       }
+        leaf name {
+          type string;
+          description
+            "A unique name for the https notif receiver.";
```
Comment (kwatsen) [inline src/yang/example-custom-module.yang]: should this be "https-notif", i.e., with the hyphen?

**Review by kwatsen — APPROVED** [review]
Minor changes requested (but approving to not block the PR)
See comments  (unsure if any are important)

**Pull request merged**